### PR TITLE
1/player branch

### PR DESCRIPTION
**Pull request by shaivikaaaa**
This pull request introduces the implementation of the Animal Selection Screen layout, which appears when the player clicks on the "Start" button. The focus of this task was on designing and implementing the visual layout with placeholder icons for the animals, ensuring a smooth transition from the main menu to this screen.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+public class AnimalSelectionScreen extends ScreenAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(AnimalSelectionScreen.class);
+    private final GdxGame game;
+    private Stage stage;
+    private Table mainTable;
+    private Image selectedAnimalImage;
+    private TextButton selectButton;
+
+    public AnimalSelectionScreen(GdxGame game) {
+        this.game = game;
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java:30]: Add logging here.

Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java:30]: Should this function be registering services here? Ie the input service to handle button clicks (I'm not sure if you should use that or the way you have done it is better - someone else can check that).

File: source/core/src/main/com/csse3200/game/GdxGame.java [hunk 2]
```diff
@@ -25,8 +26,11 @@ public void create() {
     logger.info("Creating game");
     loadSettings();
 
+    // Register the game instance with ServiceLocator
+    ServiceLocator.registerGame(this);
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/GdxGame.java:30]: Remove this - the game is not a service. See comments in MainMenuDisplay and ServiceLocator for more details.

**Review by mitsudz — CHANGES_REQUESTED** [review]
See comments. There are a number of changes requested, split into 2 main parts:
1) Refactoring of method show in AnimalSelectionScreen (it is too long with too much code duplication)
2) Changing AnimalSelectionScreen to be managed by GdxGame properly - not the ServiceLocator (the ServiceLocator doesn't need to see the game, the current changes violate the principles of the architecture that is already present).

File: source/core/src/main/com/csse3200/game/services/ServiceLocator.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.csse3200.game.services;
 
+import com.csse3200.game.GdxGame;
```
[the remark this answers is not in the document]
Comment (shaivikaaaa) [reply source/core/src/main/com/csse3200/game/services/ServiceLocator.java:3]: did not get the reason for this

**Review by lellis0406 — CHANGES_REQUESTED** [review]
Please see comments for requested changes/clarification.

**Pull request closed without merging**